python/src/equistore/operations/join.py

```diff
@@ -17,6 +17,40 @@ def join(tensormaps: List[TensorMap], axis: str):
     properties of the TensorMaps and if ``axis='samples'`` it will be the
     along the samples.
 
+    Possible clashes of the meta data like the property names
+    are resolved by the following strategy:
+
+    1. If the names (i.e property names) are the same and the
+       values (i.e property values)
+       are unique we keep the names and
+       join the values vertically. In example::
+
+        labels_1 = ["n"], [0, 2, 3]
+        labels_2 = ["n"], [1, 4, 5]
+
+      will lead to::
+
+        properties = ["n"], [0, 2, 3, 1, 4, 5]
+
+    2. If the names are the same but the values are not unique a new variable
+       "tensor" is added to the names. In example::
+
+            properties_1 = ["n"], [0, 2, 3]
+            properties_2 = ["n"], [0, 2]
+
+        will lead to::
+            properties = ["tensor_n", "n"], [[0, 0], [0, 2], [2, 0], [2, 2], [1, 2]]
```

Comment:
@agoscinski I think the extra n is a good idea but this only works nice if you have only one name like `n`. If you have two like (`n`, `l`) I think it doesn't make sense to call the attribute `tensor_n_l` because it might become super lengthy. I would stay with `tensor` for know because this is what we implemented.

Reply:
I see, I thought this can happen to multiple properties separately, but even it happens to all or none